docs/defaulting.md

```diff
@@ -0,0 +1,249 @@
+in `virt-api`. This admission controller, then reaches out to the second part,
+a special libvirt instance in a container, wrapped by a REST API, which acts as
+libvirt defaulting service. Since the defaults from this service would normally
+be filled in on the host, it is important that this controller is the last of
+all invoked admission controllers. For instance, if another admission
+controller would exist, which allows injecting Disks, like a
+[PodPreset](https://kubernetes.io/docs/tasks/inject-data-application/podpreset/)
+allows injecting volumes, it has to be invoked before this libvirt defaulting
+service. Otherwise relevant specification details, like PCI addresses might
+still be added at the host level. the final flow looks like this:
+
+```
+ client       virt-api/apiserver admission-controller 1 libvirt admission controller  libvirt defaulting service
+------------- ------------------ ---------------------- ----------------------------- ----------------------------
+
+
+POST/VM ----->  validate/auth ----> modify/validate -----> POST /apis/vms/defaults ------> fill in defaults with
+                                                                                                 libvirt
+                                                                                                    |
+                                                                                                    |
+                                                                                                    v
+                  validate <------------------------------- validate and return <--------- respond with full spec
+                      |
+                      |
+                      v
+                  persist
+                      |
+                      |
+                      v
+response <------- respond
+
+```
+
+This way of pre-filling the whole VM specification, allows users and management
+applications on top of KubeVirt, to immediately get the resulting
+specification, without the need to asynchronously watch for the missing
+defaults, to be filled in.
+
+## Important side notes
+
+### Device naming
+
+With the proposed Defaulting service, it is still pretty hard, from an
+administration and operations perspective to find out which device is which.
+libvirt at the moment does not support naming devices for the whole VM
+lifecycle (https://bugzilla.redhat.com/show_bug.cgi?id=1434451). It actually
+becomes a bigger problem with the defaulting service in place. To find out if a
+device has changed, or needs an update, you would have to rely on the order of
+the devices (which is not necessarily stable:
+https://www.redhat.com/archives/libvirt-users/2012-December/msg00087.html), or
+try to match them based on their content.
+
+### Migrating between different QEMU versions
+
+In order to make sure that the libvirt version, which does the defaulting
```

Comment:
we need to allow for up to per-VM granularity, users need to have full control when it happens. Upgrade typically does change the guest environment (e.g. machine types)
It would be reasonable to abstract this to a cluster (~ cluster level compat) or some arbitrary group of VMs in GUI / other administration tools.